frontend/src/components/filter/FilterMenu.js

```diff
@@ -6,22 +6,31 @@ import PropTypes from 'prop-types';
 import { v4 as uuidv4 } from 'uuid';
 import DropdownMenu from '../DropdownMenu';
 import FilterItem from './FilterItem';
+import { FILTER_CONFIG } from './constants';
+import { formatDateRange } from '../DateRangeSelect';
 import usePrevious from '../../hooks/usePrevious';
 
 // save this to cut down on repeated boilerplate in PropTypes
 const filterProp = PropTypes.shape({
   topic: PropTypes.string,
   condition: PropTypes.string,
-  query: PropTypes.oneOfType([PropTypes.string, PropTypes.arrayOf(PropTypes.string)]),
+  query: PropTypes.oneOfType([
+    PropTypes.string, PropTypes.arrayOf(PropTypes.string), PropTypes.number,
+  ]),
   id: PropTypes.string,
 });
 
+// a list of all the filter topics available
+const availableFilters = FILTER_CONFIG.map((f) => f.id);
+
 /**
  * Renders the entire filter menu and contains the logic for toggling it's visibility
  * @param {Object} props
  * @returns JSX Object
  */
-export default function FilterMenu({ filters, onApplyFilters }) {
+export default function FilterMenu({
+  filters, onApplyFilters, allowedFilters, dateRangeOptions,
+}) {
   const [items, setItems] = useState([...filters.map((filter) => ({ ...filter }))]);
```

Comment:
whoops sorry, missed this somehow - pretty sure that useState will create a new array but the contents will be refs not values, since they are objects, where we want fresh objects of the same value in this situation